File: BIZ/SA/report/mdx/logic/code/dsrc/MDXProcedure.java
import com.justep.report.olap.InitMDXCubeMetaDataProcedure;
import com.justep.report.olap.QueryDrillMDXProcedure;
import com.justep.report.olap.QueryMDXProcedure;


public class MDXProcedure {
	public static String queryMDX(String mdxQuery, String properties,String dataModel){
		return QueryMDXProcedure.queryMDX(mdxQuery, properties,dataModel);
	}
	
	public static String queryMDXDrill(String mdxQuery, String properties,String dataModel){
		return QueryDrillMDXProcedure.queryMDXDrill(mdxQuery, properties,dataModel);
	}
	
	public static String initMDX(String request, String restrictions ,String properties,String dataModel){
		return InitMDXCubeMetaDataProcedure.initMDX(request, restrictions, properties,dataModel);
	}
}

File: BIZ/SA/report/mdx/logic/action/mdx.procedure.m
<model xmlns="http://www.justep.com/model" xmlns:m="http://www.justep.com/model">
	
	<procedure name="queryMDXProcedure"
		code-model="/SA/report/mdx/logic/code"
		code="MDXProcedure.queryMDX">
		<parameter name="mdxQuery" type="String"/>
		<parameter name="properties" type="String"/>
		<parameter name="dataModel" type="String"/>
	</procedure>
	
	<procedure name="queryDrillMDXProcedure"
		code-model="/SA/report/mdx/logic/code"
		code="MDXProcedure.queryMDXDrill">
		<parameter name="mdxQuery" type="String"/>
		<parameter name="properties" type="String"/>
		<parameter name="dataModel" type="String"/>
	</procedure>
	
	<procedure name="initMDXCubeMetaDataProcedure"
		code-model="/SA/report/mdx/logic/code"
		code="MDXProcedure.initMDX">
		<parameter name="request" type="String"/>
		<parameter name="restrictions" type="String"/>
		<parameter name="properties" type="String"/>
		<parameter name="dataModel" type="String"/>
	</procedure>
	
</model>